### PR TITLE
Feature/theme toggle

### PR DESCRIPTION
**Pull request by vyvil**
This PR adds a way to toggle theme:

- dev way to toggle theme can now be toggled by adding value to `SHOW_THEME_TOGGLE` on env;
- there are tree options for app's theme: `Light`, `Dark` and `System's default`. If later one is selected, theme gets updated if user changes mode on his/her device.
- on first open, an alert is shown to those users that are set dark theme on their phones showing that dark mode is available on the wallet as well.

PR can be reviewed now. Yet it would be better to merge that in when all themes related PRs are in.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -163,43 +192,49 @@ class App extends React.Component<Props, *> {
...
+        <ThemeProvider theme={theme}>
+          <React.Fragment>
+            <Root>
+              <RootNavigation
+                ref={(node) => {
+                  if (!node) return;
+                  setTopLevelNavigator(node);
+                }}
+              />
+              {!!SHOW_THEME_TOGGLE &&
```
Comment (grilix) [inline App.js]: :clap:

File: App.js [hunk 2]
```diff
@@ -150,6 +159,26 @@ class App extends React.Component<Props, *> {
...
+        'Would you like to turn on Dark mode now? You can always switch between modes in settings.',
+        [
+          { text: 'Cancel', onPress: () => markThemeAlertAsShown() },
+          {
+            text: 'Turn on',
+            onPress: () => changeAppTheme(DARK_THEME),
+          },
+        ],
+      );
+    }
```
Comment (poocart) [inline App.js]: I know these kind of comments are annoying, but maybe just `if (defaultPreference !== DARK_PREFERENCE) return`? 😄

File: src/actions/appActions.js [hunk 3]
```diff
@@ -224,12 +223,18 @@ export const initAppAndRedirectAction = (appState: string, platform: string) =>
 
       // check if current user has theme set and set it to default if
       const hasTheme = get(appSettings, 'themeType');
+      // check if theme is set to system's default
+      const isThemeSetAsSystemDefault = get(appSettings, 'isSetAsSystemPrefTheme');
```
Comment (poocart) [inline src/actions/appActions.js]: How about moving this to line 234 where the logic happens?

File: src/actions/walletActions.js [hunk 4]
```diff
@@ -39,19 +39,21 @@ import shuffle from 'shuffle-array';
...
-  return async (dispatch: Function, getState: () => Object, api: Object) => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
     try {
       const importedWallet = ethers.Wallet.fromMnemonic(tWordsPhrase);
 
       api.init();
       let apiUser = {};
       const addressValidationResponse = await api.validateAddress(importedWallet.address);
-      if (addressValidationResponse.walletId) {
+      if (!addressValidationResponse.error) {
```
Comment (poocart) [inline src/actions/walletActions.js]: I assume you carefully checked with this one as we had issues here.

File: src/screens/Settings/Settings.js [hunk 5]
```diff
@@ -465,12 +527,13 @@ class Settings extends React.Component<Props, State> {
     );
   };
 
-  renderListItem = (field: string, onSelect: Function) => ({ item: { name } }: Object) => {
+  renderListItem = (field: string, onSelect: Function, currentValue) => ({ item: { name, value } }: Object) => {
```
Comment (poocart) [inline src/screens/Settings/Settings.js]: maybe you could add `currentValue` type as well?

File: App.js (hunk 2 above)
Comment (alazarevski) [reply App.js]: I'd say it's ok without it on functions with less than 20 lines of code

**Pull request merged**